### PR TITLE
task: create and use typography theming

### PR DESCRIPTION
**Pull request by tborres**
- update typography variables to match the zeplin styleguide
- update the font variants
- update the app to use `rem`s instead of `px` and variables when possible

Closes [B2-66](https://b2io.atlassian.net/browse/B2-66)

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ module.exports = {
         ...config.resolve,
         alias: {
           ...config.resolve.alias,
+          '~': toPath('src/'),
```
Comment (tborres) [inline .storybook/main.js:15]: 💬 Fixes issue when importing via `~` in files that storybook touches.

File: .storybook/preview.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ export const decorators = [
           * {
             box-sizing: border-box;
             font-family: Roobert;
-            font-feature-settings: 'ss02';
+            font-feature-settings: 'ss02', 'ss05';
```
Comment (tborres) [inline .storybook/preview.js:26]: 💬 Fancy quote. I had visually checked and compared each of the variations to the design files.

File: src/components/CTA.tsx [hunk 3]
```diff
@@ -8,19 +8,16 @@ export type CTAProps = {
 };
 
 const A = styled.a(({ theme }) => ({
-  color: theme.colors.text,
+  ...theme.textVariants.CTA,
```
Comment (tborres) [inline src/components/CTA.tsx:11]: 💬 I'm interested to hear if there is a better way to use the `textVariants` than this.

File: src/components/Header.tsx [hunk 4]
```diff
@@ -59,24 +59,35 @@ const Nav = styled.nav`
...
     }
   }
+`;
 
-  a {
-    color: ${(props) => props.theme.colors.offWhite};
-    font-size: 17px;
-    line-height: 28px;
-    text-decoration: none;
+const A = styled.a(({ theme }) => ({
```
Comment (tborres) [inline src/components/Header.tsx:72]: 💬 I switched to an object-based styled component to be able to spread the textVariant styling. I'm not sure if there is a better / more appropriate way to do this.

File: src/theme/typography.ts [hunk 5]
```diff
@@ -1,57 +1,92 @@
...
 
 export const textVariants = {
   body: {
     color: colors.text,
-    fontSize: '17px',
-    letterSpacing: '0px',
-    lineHeight: '28px',
+    fontSize: '1.063rem',
+    fontVariationSettings: '"wght" 500',
+    letterSpacing: '0.006875rem',
```
Comment (tborres) [inline src/theme/typography.ts:9]: 💬 This (and similar) letter spacing is per the Zeplin styleguide, but it doesn't seem that it's actually being used in the designs (or the designs simply round it down to 0).

File: .storybook/preview.js (hunk 2 above)
Comment (k8hansen) [reply .storybook/preview.js:26]: Wasn't entirely sure how to test this, but I'm seeing a fancy 'g' in storybook and it runs so I consider it a pass.

**Review by k8hansen — APPROVED** [review]
:shipit:

File: .storybook/preview.js (hunk 2 above)
Comment (tborres) [reply .storybook/preview.js:26]: `ss02` = fancy g
`ss05` = fancy quote.
☝️ technical terms

**Comment by tborres** [comment]
I'm going to merge this in now so that it can start being used, but I'm still interested to hear feedback from anybody who's got some.

**Pull request merged**